### ISSUE TITLE
group membership check in case of LDAP authentication

### ISSUE THREAD

**Issue by JanuszU** [vinyldns/vinyldns#929]
Please consider adding a group membership check in case of LDAP authentication, it is much more flexible than just narrowing users to OU

**Comment by JanuszU** [comment]
a simple solution:
1. in ../vinyldns/docker/portal/application.conf  in section "LDAP", add a new config parameter
    groupDN = "CN=vinyl_admins,OU=ourDept,DC=company,DC=com,DC=pl"
2. in ../vinyldns/modules/portal/app/controllers/Settings.scala add a line:
    val ldapGroupDN: String = config.get[String]("LDAP.groupDN")
3. in ../vinyldns/modules/portal/app/controllers/LdapAuthenticator.scala  replace a line
old line (line# 129):
      s"(${settings.ldapUserNameAttribute}=$lookupUserName)",
new line:
      s"(&(objectClass=user)(${settings.ldapUserNameAttribute}=$lookupUserName)(memberOf=${settings.ldapGroupDN}))",

Enjoy ;)
PS. sorry , no group recursion

**Comment by pauljamescleary** [comment]
@JanuszU yes this should be possible.  If the `groupDN` is missing, works like it does today, if present it will be used.  Sounds pretty straightforward
